### ISSUE TITLE
Fields filter doesn't limit fields from mongo

### ISSUE THREAD

**Issue by Gonyoda** [strongloop/loopback-connector-mongodb#435]
# Description/Steps to reproduce

The 'fields' filter option to loopback queries isn't used properly in the `find` call to the mongo driver.  This means the entire mongo document(s) is/are returned to the Node client.  Loopback then culls the properties from the objects in memory before returning them to the calling client.  The unit test called `all return should honor filter.fields` passes because of [loopback](https://github.com/strongloop/loopback-datasource-juggler/blob/ef3191d3e55466d216dab8a1f7e5c32e1b33431c/lib/dao.js#L2023), not because of the mongodb.js code.

Other than performing log statements in all's data.map callback, I don't know how to demonstrate the problem or create a meaningful unit test.  The problem is solved by passing an object with a `projection` property of the fields to the `this.execute` call. See https://github.com/Snagajob/loopback-connector-mongodb/pull/1/commits/1dea3c25a61e93dad2aa546f472dfe0ab541d014

Instead of an array of fields:
```[ 'field1', 'field2', 'field3' ]```

It should pass an object of:
```
{
  projection: {
    field1: 1, 
    field2: 1, 
    field3: 1
  }
}
```

# Link to reproduction sandbox

No link.

# Expected result

That the `fields` filter is applied to the mongo query.

# Additional information

Here's a PR to our fork to see the changes necessary.  I can clean up the commits and move this PR to this repo if it's OK: https://github.com/Snagajob/loopback-connector-mongodb/pull/1

**Comment by Gonyoda** [comment]
Here is output from the unit test passing an array to mongo.find:
```
  mongodb connector
from mongo:
{ _id: 5b157bf95200cb3958c17f44, title: 'b', content: 'BBB' }
in unit test:
{ title: 'b' }
    √ all return should honor filter.fields (56ms)
```

Here is the output after passing an object with projection property:
```
  mongodb connector
from mongo:
{ _id: 5b157c422ffe6e2ae88c7248, title: 'b' }
in unit test:
{ title: 'b' }
    √ all return should honor filter.fields (46ms)
```

**Comment by raymondfeng** [comment]
Hmm, good catch. It seems that mongodb@3.x remove the support of `fields` as a separate argument. I wonder why the test cases don't catch that.

Can you submit a PR?

Please note `options.projection` also allows the array - See https://github.com/mongodb/node-mongodb-native/blob/master/lib/collection.js#L273

**Comment by dhmlau** [comment]
PR #436 has landed. It's available in `loopback-connector-mongodb@3.4.4`.
